src/libsyntax/ext/mtwt.rs

```diff
@@ -171,15 +104,24 @@ pub fn outer_mark(ctxt: SyntaxContext) -> Mrk {
     })
 }
 
+/// If `ident` is macro expanded, return the source ident from the macro definition
```

Comment:
Currently, this function returns `None` for idents expanded from procedural macros.

Ideally, I think it should return the unmarked ident along with a "null" macro definition mark (since procedural macros are not expanded into existence).